client/rest/src/main/java/org/elasticsearch/client/DeadHostState.java

```diff
@@ -61,11 +74,28 @@ long getDeadUntilNanos() {
         return deadUntilNanos;
     }
 
+    int getFailedAttempts() {
+        return failedAttempts;
+    }
+
+    @Override
+    public int compareTo(DeadHostState other) {
+        return Long.compare(deadUntilNanos, other.deadUntilNanos);
+    }
+
     @Override
     public String toString() {
         return "DeadHostState{" +
                 "failedAttempts=" + failedAttempts +
                 ", deadUntilNanos=" + deadUntilNanos +
                 '}';
     }
+
+    static class TimeSupplier {
```

Comment:
Can you make this an interface? I think it'd *feel* cleaner that way.

And it is probably worth javadoc to explain that how you fetch time is pluggable for testing. Lots of folks are used to this but anyone that hasn't had to write unit tests for code involving timeouts and such won't be used to it.

Reply:
will do, I miss default methods here :)